src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,122 @@
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(nullLoginUser));
+    }
+
+    @Test
+    void register_nullPassword_NotOk() {
+        User nullPasswordUser = new User("validlogin", null, 20);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(nullPasswordUser));
+    }
+
+    @Test
+    void register_nullAge_NotOk() {
+        User nullAgeUser = new User("validlogin", "validpassword", null);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(nullAgeUser));
+    }
+
+    @Test
+    void register_shortLogin_NotOk() {
+        User shortLoginUser = new User("login", "validpassword", 20);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(shortLoginUser));
+    }
+
+    @Test
+    void register_shortPassword_NotOk() {
+        User shortPasswordUser = new User("validlogin", "12345", 20);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(shortPasswordUser));
+    }
+
+    @Test
+    void register_ageLessThan18_NotOk() {
+        User youngUser = new User("validlogin", "validpassword", 17);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(youngUser));
+    }
+
+    @Test
+    void register_sameLogin_NotOk() {
+        User sameLoginUser = new User("mateacademy", "password", 20);
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(sameLoginUser));
+    }
+
+    @Test
+    void register_addValidUser_Ok() {
+        User validUser = new User("validlogin", "validpassword", 25);
+        registrationService.register(validUser);
+        assertEquals(validUser, storageDao.get(validUser.getLogin()));
+        assertEquals(2, Storage.people.size());
+    }
+
+    @Test
+    void register_addMinAgeLoginPassword_Ok() {
+        User minAgeLoginPasswordUser = new User("loginn", "passwo", 18);
+        registrationService.register(minAgeLoginPasswordUser);
+        assertEquals(minAgeLoginPasswordUser, storageDao.get(minAgeLoginPasswordUser.getLogin()));
+        assertEquals(2, Storage.people.size());
+    }
+
+    @Test
+    void register_addValidUsers_Ok() {
+        User firstUser = new User("firstlogin", "firstpassword", 25);
+        User secondUser = new User("secondlogin", "secondpassword", 18);
+        User thirdUser = new User("thirdlogin", "thirdpassword", 80);
+        registrationService.register(firstUser);
+        registrationService.register(secondUser);
+        registrationService.register(thirdUser);
+        assertEquals(firstUser, storageDao.get(firstUser.getLogin()));
+        assertEquals(secondUser, storageDao.get(secondUser.getLogin()));
+        assertEquals(thirdUser, storageDao.get(thirdUser.getLogin()));
+        assertEquals(4, Storage.people.size());
```

Comment:
The test correctly registers multiple valid users and checks that they are added to the storage. Ensure that the storage size is correctly verified.